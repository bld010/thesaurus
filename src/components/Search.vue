<template>
  <form class="Search">
    <label for="search">Search for synonyms</label>
    <input
      placeholder="Search ..."
      name="search"
    >
    <button v-on:click.prevent>Search</button>
  </form>
</template>

<script>


export default {
  name: 'Search'
}

</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

form {
  margin-top: 20px;
  display: flex;
  flex-direction: row;
  min-width: 30%;
  max-width: 400px;
  margin: 20px auto 20px auto;
  padding-bottom: 20px;
  border-bottom: 1px solid #039B77;
  justify-content: center;
}

input {
  margin-left: 5px;
}
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
