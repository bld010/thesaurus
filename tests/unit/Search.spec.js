import Search from '../../src/components/Search.vue';
import { shallowMount } from '@vue/test-utils';
import '@vue/cli-plugin-unit-jest'

describe('Search', () => {
  it('should match the snapshot', () => {
    const wrapper = shallowMount(Search);

    expect(wrapper).toMatchSnapshot();
  })

  it('should set the correct default data', () => {

    const expected = {
      searchTerm: '',
      results: [],
      synonyms: [],
      error: '',
      loading: false
    }

    expect(typeof Search.data).toBe('function');
    const defaultData = Search.data();
    expect(defaultData).toEqual(expected)
  })
})