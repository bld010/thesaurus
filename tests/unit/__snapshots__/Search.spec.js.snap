// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
<div>
  <form class="Search"><label for="search">Search for synonyms</label> <input placeholder="Search ..." name="search"> <button>Search</button></form>
  <div>
    <!---->
  </div>
</div>
`;
